### ISSUE TITLE
[Bug] Is the MoE expert routing in _sample_timesteps intended to be flow-shift-aware?

### ISSUE THREAD

**Issue by Xiaoyuan-Isaac-Wang** [hao-ai-lab/FastVideo#1370]
### Describe the bug

I'm fine-tuning Wan2.2-T2V-A14B with the default `flow_shift=12.0` and `boundary_ratio=0.875`, and I'm trying to make sure I understand the intended behavior of the MoE expert selection in [`training_pipeline._sample_timesteps`](https://github.com/hao-ai-lab/FastVideo/blob/17f07bc313d77adaf1d60aa344d92f8145e54cfb/fastvideo/training/training_pipeline.py#L309-L341). Sorry if I'm missing something obvious.
My current understanding:
- During inference, [`WanPipeline`](https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/wan/pipeline_wan.py#L577-L595) routes purely on the actual t value (`if t >= boundary_ratio * num_train_timesteps`), i.e. the boundary lives in **t-space**.
- During training, `_sample_timesteps` (a) flips a Bernoulli at probability `boundary_ratio` to pick an expert, then (b) rescales `u` to the corresponding sub-interval using `1 - boundary_ratio` as the split point in **u-space**.
For `flow_shift=1`, u and t are linearly related, so the t-space boundary at `875` and the u-space split at `0.125` line up. For `flow_shift > 1`, however, the scheduler's `timesteps[]` is non-linearly warped at init, and the same u value maps to a different t:

| `flow_shift` | u = 0.125 corresponds to t ≈ | t = 875 corresponds to u ≈ |
|---|---|---|
| 1 | 875 | 0.125 |
| 5 | 955 | 0.417 |
| 12 | **988** | **0.632** |

If I'm reading the code right, with the official `flow_shift=12.0` setting from [`wan_t2v_A14B.py`](https://github.com/Wan-Video/Wan2.2/blob/main/wan/configs/wan_t2v_A14B.py) (`sample_shift = 12.0`):
- `transformer` (high-noise expert) is trained on `u ∈ [0, 0.125]` → `t ∈ [988, 1000]`, but at inference it's called for the full `t ∈ [875, 1000]` (which is ~64% of inference steps under `shift=12`).
- `transformer_2` is trained on `u ∈ [0.125, 1]` → `t ∈ [12, 988]`, so it sees most of the `[875, 988]` band, which inference routes away from it.
- Bernoulli probability is `boundary_ratio = 0.875`, so `transformer_2` gets ~87.5% of opt steps even though inference uses it for only ~36% of steps under `shift=12`.
A couple of questions:
1. Is this intentional? E.g. is `boundary_ratio` meant to be interpreted purely as a u-space split / training-time budget knob (rather than a t-space boundary that mirrors the inference check), with the assumption that the experts will generalize across t via the timestep embedding?
2. If a t-space alignment **is** the intended semantic (matching inference), would something like the following be a reasonable fix? At init, search the (already-shifted) `noise_scheduler.timesteps` for the index closest to `boundary_timestep`, and use that to set both the u-split and the Bernoulli probability:
```python
target_t = self.training_args.boundary_ratio * self.noise_scheduler.config.num_train_timesteps
u_boundary = (self.noise_scheduler.timesteps - target_t).abs().argmin().item() / num_train_timesteps
# u_boundary == 1 - boundary_ratio for shift=1 (backward-compatible);
# u_boundary ≈ 0.632 for shift=12 (lines up with inference's t-routing)
```
VideoX-Fun seems to take a similar approach by doing the boundary check directly in idx-space on the shifted `timesteps[]`, see [`scripts/wan2.2/train.py#L1504-L1521`](https://github.com/aigc-apps/VideoX-Fun/blob/199a43b544d5db4109bc1de4dd9bd029b6d25cff/scripts/wan2.2/train.py#L1504-L1521).

A related minor thing in the same function: when the high-noise expert is selected, the `elif self.transformer_2 is not None: u = u * (1 - boundary_ratio)` branch is currently commented out (originally as a "patch for now to align with non-MoE timestep logic" in PR #818). Is that still the intended behavior, or was it meant to be re-enabled at some point? With it commented out, `transformer` is trained on the full `u ∈ [0, 1]` instead of the `[0, 1 - boundary_ratio]` sub-interval the rest of the code seems to assume.
Happy to send a PR for either if it's useful, just wanted to check the intent first.

### Reproduction

Wan2.2-T2V-A14B full fine-tune, flow_shift=12.0, boundary_ratio=0.875, weighting_scheme=uniform, FSDP+SP

### Environment

cuda 13.1
